File: app/src/main/java/com/example/networkaplication/persistance/repository/database/DatabaseRepositoryImpl.java
package com.example.networkaplication.persistance.repository.database;

import android.content.Context;
import android.support.annotation.VisibleForTesting;

import com.example.networkaplication.persistance.repository.database.dao.DetailsDao;
import com.example.networkaplication.persistance.repository.database.dao.MovieQueryDao;

public class DatabaseRepositoryImpl implements DatabaseRepository {
    private AppDatabase db;

    public DatabaseRepositoryImpl(Context context) {
        db = AppDatabase.build(context);
    }

    @Override
    public MovieQueryDao movieQueryDao() {
        return db.movieQueryDao();
    }

    @Override
    public DetailsDao detailsDao() { return db.detailsDao(); }

    @VisibleForTesting
    public void setDb(AppDatabase db) {
        this.db = db;
    }
}

File: app/src/main/java/com/example/networkaplication/persistance/repository/database/DatabaseRepositoryImpl.kt
package com.example.networkaplication.persistance.repository.database

import android.content.Context

import androidx.annotation.VisibleForTesting

import com.example.networkaplication.persistance.repository.database.dao.DetailsDao
import com.example.networkaplication.persistance.repository.database.dao.MovieQueryDao

class DatabaseRepositoryImpl(context: Context) : DatabaseRepository {
    private var db: AppDatabase? = null

    init {
        db = AppDatabase.build(context)
    }

    override fun movieQueryDao(): MovieQueryDao {
        return db!!.movieQueryDao()
    }

    override fun detailsDao(): DetailsDao {
        return db!!.detailsDao()
    }

    @VisibleForTesting
    fun setDb(db: AppDatabase) {
        this.db = db
    }
}
